src/python/zquantum/core/estimation/_estimation.py

```diff
@@ -225,6 +225,88 @@ def evaluate_estimation_circuits(
     ]
 
 
+def split_constant_estimation_tasks(
+    estimation_tasks: List[EstimationTask],
+) -> Tuple[List[EstimationTask], List[EstimationTask], List[int], List[int]]:
+    """This function splits a give list of EstimationTask into two: one that
+    contains EstimationTasks that only contain constants, and one that contains
+    EstimationTasks that should actually be submitted to a backend. EstimationTask
+    that require 0 shot generate an error.
+
+    Args:
+        estimation_tasks (List[EstimationTask]): The list of estimation tasks for which
+                         Expectation Values are wanted.
+
+    Returns:
+        estimation_tasks_to_measure (List[Circuit]): A new list of estimation tasks that only
+                                     contains the ones that should actually be submitted to the
+                                     backend
+        estimation_tasks_for_constants (List[Circuit]): A new list of estimation tasks that
+                                        contains the EstimationTasks with only constant terms
+        indices_to_measure (List[int]): A list containing the indices of the EstimationTasks we will
+                            actually measure, i.e. the ith estimation_tasks_to_measure expectation
+                            value will go into the indices_to_measure[i] position.
+        indices_for_constants (List[int]): A list containing the indices of the EstimationTasks for
+                               constant terms.
+    """
+
+    estimation_tasks_to_measure = []
+    estimation_tasks_for_constants = []
+    indices_to_measure = []
+    indices_for_constants = []
+    for i, e in enumerate(estimation_tasks):
```

Comment:
I would change it to `i, task` ;)

Reply:
Addressed in be2a71d78c59d2ad544bc9a23dacd3f5a7dad106